src/controllers/deliveryLog.js

```diff
@@ -1,24 +1,32 @@
 import { sendDataResponse } from '../utils/responses.js'
+import DeliveryLog from '../domain/deliveryLog.js'
+import DeliveryLogLines from '../domain/deliveryLogLines.js'
 
 export const create = async (req, res) => {
   const { date, cohort_id: cohortId, lines } = req.body
+  const userId = req.user.id
+  const newDeliveryLogData = {
+    date: new Date(date),
+    cohortId,
+    userId
+  }
 
-  return sendDataResponse(res, 201, {
-    log: {
-      id: 1,
-      cohort_id: cohortId,
-      date,
-      author: {
-        id: req.user.id,
-        first_name: req.user.firstName,
-        last_name: req.user.lastName
-      },
-      lines: lines.map((line, index) => {
-        return {
-          id: index + 1,
-          content: line.content
-        }
-      })
-    }
-  })
+  const deliveryLogToCreate = await DeliveryLog.fromJson(newDeliveryLogData)
+  const deliveryLog = await deliveryLogToCreate.save()
+
+  const deliveryLogLines = []
+  const postId = deliveryLog.id
+
+  for (const line of lines) {
```

Comment:
Use `map()` instead. Something like:

```js
const deliveryLogLines = lines.map(line => {
    const data = { content: line.content, logId: deliveryLog.id }
    return await DeliveryLogLines.fromJson(data)
})
```

You could then also save it during the loop:

```js
const deliveryLogLines = lines.map(line => {
    const data = { content: line.content, logId: deliveryLog.id }
    const logLineToCreate =  await DeliveryLogLines.fromJson(data)
    return await logLineToCreate.save()
})
```

Reply:
Share your solution here first - if it's reasonable, yes let's map.

Reply:
> Should I pass this to a current member of the teacher admin team?

@Alex90Jennings Yes

Reply:
The implemmentation is now using a map as suggested.